From 231a632a294bb303d7d50ccc1d7650526493efa7 Mon Sep 17 00:00:00 2001
From: Mike Bayer <mike_mp@zzzcomputing.com>
Date: Wed, 23 Nov 2022 17:36:42 -0500
Subject: [PATCH] repair register opt calls

flake8 uses only argparse option styles now, use
str instead of "string" for types given.

Fixes: #189
---
 flake8_import_order/flake8_linter.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/flake8_import_order/flake8_linter.py b/flake8_import_order/flake8_linter.py
index c6fb1ba..7734592 100644
--- flake8_import_order/flake8_linter.py
+++ flake8_import_order/flake8_linter.py
@@ -25,7 +25,7 @@ def add_options(cls, parser):
             "--application-import-names",
             default="",
             action="store",
-            type="string",
+            type=str,
             help="Import names to consider as application-specific",
             parse_from_config=True,
             comma_separated_list=True,
@@ -35,7 +35,7 @@ def add_options(cls, parser):
             "--application-package-names",
             default="",
             action="store",
-            type="string",
+            type=str,
             help=("Package names to consider as company-specific "
                   "(used only by 'appnexus' style)"),
             parse_from_config=True,
@@ -46,7 +46,7 @@ def add_options(cls, parser):
             "--import-order-style",
             default=DEFAULT_IMPORT_ORDER_STYLE,
             action="store",
-            type="string",
+            type=str,
             help=("Style to follow. Available: " +
                   ", ".join(cls.list_available_styles())),
             parse_from_config=True,
